core/src/test/java/org/infinispan/test/TestingUtil.java

```diff
@@ -1772,4 +1774,16 @@ public static boolean isTriangleAlgorithm(CacheMode cacheMode, boolean transacti
    public static <K,V> Map.Entry<K,V> createMapEntry(K key, V value) {
       return new AbstractMap.SimpleEntry<>(key, value);
    }
+
+   public static void inject(Object instance, Object... components) {
```

Comment:
Maybe just add a comment that this only works on fields with `@Inject` and doesn't invoke any methods with `@Inject`.

This is also much cleaner with fields since we can just override specific components :+1:

Reply:
Injecting multiple named executors wouldn't work that well, maybe throw an exception when 2 or more components match the same field? The test could create a proper `ComponentRegistry` and call `wireDependencies` in those cases.

Reply:
@danberindei It's not bullet proof, I did not bother since this is only a testing helper. When you need it, you'll code that, and test that such util test code works.

Reply:
I've added the check for multiple matches.